tests/migrations/test_commands.py

```diff
@@ -338,6 +339,127 @@ def test_showmigrations_plan_squashed(self):
+        out = six.StringIO()
+        call_command("showmigrations", "mutate_state_b", "author_app", format='plan', stdout=out)
+        self.assertEqual(
+            "[ ]  author_app.0001_initial\n"
+            "[ ]  book_app.0001_initial\n"
+            "[ ]  author_app.0002_alter_id\n"
+            "[ ]  mutate_state_b.0001_initial\n"
+            "[ ]  mutate_state_b.0002_add_field\n",
+            out.getvalue()
+        )
+        # Multiple apps: args order shouldn't matter (the same result is expected as above).
+        out = six.StringIO()
+        call_command("showmigrations", "author_app", "mutate_state_b", format='plan', stdout=out)
+        self.assertEqual(
+            "[ ]  author_app.0001_initial\n"
+            "[ ]  book_app.0001_initial\n"
+            "[ ]  author_app.0002_alter_id\n"
+            "[ ]  mutate_state_b.0001_initial\n"
+            "[ ]  mutate_state_b.0002_add_field\n",
+            out.getvalue()
+        )
+
+    @override_settings(INSTALLED_APPS=[
+        "migrations.migrations_test_apps.mutate_state_a",
+        "migrations.migrations_test_apps.mutate_state_b",
+        "migrations.migrations_test_apps.alter_fk.author_app",
+        "migrations.migrations_test_apps.alter_fk.book_app",
+    ])
+    def test_showmigrations_plan_app_label_verbose(self):
+        """
+        Makes sure that `showmigrations --plan app_label -v 2`
+        output is verbose enough.
+        """
+        out = six.StringIO()
+        call_command("showmigrations", "author_app", "mutate_state_b", format='plan', stdout=out, verbosity=2)
+        self.assertEqual(
+            "[ ]  author_app.0001_initial\n"
+            "[ ]  book_app.0001_initial ... (author_app.0001_initial)\n"
+            "[ ]  author_app.0002_alter_id ... (author_app.0001_initial, book_app.0001_initial)\n"
+            "[ ]  mutate_state_b.0001_initial\n"
+            "[ ]  mutate_state_b.0002_add_field ... (mutate_state_b.0001_initial)\n",
+            out.getvalue()
+        )
+
+        # mutate_state_a app should be ignored even while it's in INSTALLED_APPS.
+        self.assertNotIn("mutate_state_a", out.getvalue())
+        self.assertIn("migrations.migrations_test_apps.mutate_state_a", settings.INSTALLED_APPS)
+
+    @override_settings(INSTALLED_APPS=["migrations.migrations_test_apps.unmigrated_app"])
+    def test_showmigrations_plan_app_label_error(self):
+        """
+        Makes sure that `showmigrations --plan app_label` raises error
+        when no app or no migrations are present in provided app labels.
+        """
+        # App with no migrations.
+        with self.assertRaisesMessage(CommandError, "No migrations present for: unmigrated_app"):
+            call_command("showmigrations", "unmigrated_app", format='plan')
+
+        # Non existing app (wrong app label).
+        with self.assertRaisesMessage(CommandError, "No migrations present for: duth2016"):
+            call_command("showmigrations", "duth2016", format='plan')
```

Comment:
It would help readability to use a name like "nonexistent_app" rather than "duth..".